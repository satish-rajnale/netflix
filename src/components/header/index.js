import { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Profile,
  PlayButton,
  Picture,
  Dropdown,
  Group,
  Link,
  FeatureCallOut,
  Text,
  Background,
  ButtonLink,
  Container,
  Logo,
  Feature,
  SearchInput,
  SearchIcon,
  Search,
} from "./styles/header";
import React from 'react';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png" alt="search" />
      </SearchIcon>
      <SearchInput
        placeholder="Search films & series"
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
      />
    </Search>
  );
};
