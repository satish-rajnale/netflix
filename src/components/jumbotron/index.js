import React from 'react';
import {Inner} from './styles/jumbotron'

export default function Jumbotron({children, direction = 'row', ...restProps}){
    return (
        <Inner direction={direction}>
            <p>Hello again</p>
        </Inner>
    )
}


Jumbotron.Container = function JumbotronContainer({ children, ...restProps}){
    return <Container {...restProps}>{children}</Container>
}