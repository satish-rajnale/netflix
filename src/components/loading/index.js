import { LockBody, Spinner, Picture, ReleaseBody } from "./styles/loading";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingReleasebody() {
  return <ReleaseBody />;
};
