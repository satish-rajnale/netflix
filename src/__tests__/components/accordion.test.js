import React from "react";
import { render, fireEvent } from "@testing-library/react";
import faqsData from "../../fixtures/faqs.json";
import { Accordion } from "../../components";

describe("<Accordion/>", () => {
  it("renders <Accordion/> with data", () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Item>
          {" "}
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Item>
      </Accordion>
    );

    expect(getByText("Frequently Asked Questions")).toBeTruthy();
    expect(getByText("What is Netflix?")).toBeTruthy();
    expect(getByText("How much does Netflix cost?")).toBeTruthy();
    expect(getByText("Where can I watch?")).toBeTruthy();
   
    expect(getByText("How do I cancel?")).toBeTruthy();
    expect(getByText("What can I watch on Netflix?")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();

  
});

it("opens and closes <Accordion/> component", () => {
    const { container, queryByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Item>
          {" "}
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Item>
      </Accordion>
    );

    const netflixBodyText = 
    "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!" ;

    expect(queryByText(netflixBodyText)).toBeFalsy();
    fireEvent.click(queryByText("What is Netflix?"));
    expect(queryByText(netflixBodyText)).toBeTruthy();

    fireEvent.click(queryByText("What is Netflix?"));
    expect(queryByText(netflixBodyText)).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();

});



});
