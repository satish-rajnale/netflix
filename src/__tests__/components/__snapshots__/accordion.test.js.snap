// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion/> opens and closes <Accordion/> component 1`] = `
<div
  class="accordion__Container-l3iym9-0 jhRdpM"
>
  <div
    class="accordion__Inner-l3iym9-1 buOkWn"
  >
    <h1
      class="accordion__Title-l3iym9-2 luxaO"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-l3iym9-3 gnmany"
    >
       
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          What is Netflix?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          How much does Netflix cost?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          Where can I watch?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          How do I cancel?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          What can I watch on Netflix?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion/> renders <Accordion/> with data 1`] = `
<div
  class="accordion__Container-l3iym9-0 jhRdpM"
>
  <div
    class="accordion__Inner-l3iym9-1 buOkWn"
  >
    <h1
      class="accordion__Title-l3iym9-2 luxaO"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-l3iym9-3 gnmany"
    >
       
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          What is Netflix?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          How much does Netflix cost?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          Where can I watch?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          How do I cancel?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-3 gnmany"
      >
        <div
          class="accordion__Header-l3iym9-4 kjSZAf"
        >
          What can I watch on Netflix?
          <img
            alt="open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
